### PR TITLE
TEAMNADO-2277 Hide SCA Toggle display when administratively disabled

### PR DESCRIPTION
**Pull request by joeldmyers**
## What's Included: 

- The app now uses react-query, rather than useContext, to call the authentication APIs and manage user permissions state.
- useUserPermissions hook now calls two APIs - the cloud dot API method (`window.insights.chrome.auth.getUser()`) and the new API call to `/management/v1/organization` to get the SCA capable status. It then stores org admin status and SCA capable status in the react-query cache to be consumed through the app (via the `useQueryClient` method from react-query).
- The SatelliteManifestPanel has been updated to conditionally hide the SCA toggle column if the user is not SCA enabled.
- Related tests that relied on old useContext user data have been updated
- This also includes a new test helper to create a new react-query wrapper for each test, to isolate the tests (this was causing issues in the past, such as not being able to hit isError in a test, because the isSuccess result of a previous test was being cached in the react-query cache).

### CODE REVIEW

File: src/components/SatelliteManifestPanel/SatelliteManifestPanel.tsx [hunk 1]
```diff
@@ -49,22 +49,32 @@ interface BooleanDictionary {
...
-        <>
-          Simple Content Access
-          <SCAInfoIconWithPopover />
-        </>
-      ),
-      transforms: [sortable, cellWidth(20)]
-    },
-    { title: 'UUID', transforms: [sortable] }
-  ]);
+  const getColumns = (userPermissions: UserPermissions) => {
```
Comment (zanewoodfin) [inline src/components/SatelliteManifestPanel/SatelliteManifestPanel.tsx:54]: Doesn't this function have direct access to the userPermissions prop?  Why did you add it as an arg to the method?

File: src/components/SatelliteManifestPanel/SatelliteManifestPanel.tsx [hunk 2]
```diff
@@ -111,13 +121,13 @@ const SatelliteManifestPanel: FunctionComponent<SatelliteManifestPanelProps> = (
     setDetailsDrawerIsExpanded(false);
   };
 
-  const formatRow = (row: string[], rowIndex: number) => {
+  const formatRow = (row: string[], rowIndex: number, userPermissions: UserPermissions) => {
```
Comment (zanewoodfin) [inline src/components/SatelliteManifestPanel/SatelliteManifestPanel.tsx:124]: Same as above, why'd you add userPermissions as an arg?

File: src/components/SatelliteManifestPanel/SatelliteManifestPanel.tsx [hunk 3]
```diff
@@ -49,22 +49,32 @@ interface BooleanDictionary {
...
+    ];
+
+    if (userPermissions.isSCACapable === false) {
+      // remove SCA Status column
+      columns.splice(2, 1);
+    }
+    return columns;
+  };
+
+  const columns = getColumns(userPermissions);
```
Comment (zanewoodfin) [inline src/components/SatelliteManifestPanel/SatelliteManifestPanel.tsx:77]: Can you just call the getColumns function directly in the `cells={columns}` location below instead of having this extraneous variable?  Then you could move the function down below all this state stuff with the other functions.

Maybe we should also rename the getColumns function because it's not very clear that it's just metadata about the columns.  Of course the prop it's being passed to in Table also isn't clear, but there's nothing to do about that.

**Review by zanewoodfin — CHANGES_REQUESTED** [review]

File: src/components/SatelliteManifestPanel/SatelliteManifestPanel.tsx [hunk 4]
```diff
@@ -129,6 +139,11 @@ const SatelliteManifestPanel: FunctionComponent<SatelliteManifestPanelProps> = (
       </React.Fragment>,
       uuid
     ];
+    if (userPermissions.isSCACapable === false) {
```
[the remark this answers is not in the document]
Comment (joeldmyers) [reply src/components/SatelliteManifestPanel/SatelliteManifestPanel.tsx]: Aha ok I put in a fix below.

File: src/components/SatelliteManifestPanel/SatelliteManifestPanel.tsx [hunk 5]
```diff
@@ -49,22 +49,32 @@ interface BooleanDictionary {
...
+            Simple Content Access
+            <SCAInfoIconWithPopover />
+          </>
+        ),
+        transforms: [sortable, cellWidth(20)]
+      },
+      { title: 'UUID', transforms: [sortable] }
+    ];
+
+    if (userPermissions.isSCACapable === false) {
```
[the remark this answers is not in the document]
Comment (joeldmyers) [reply src/components/SatelliteManifestPanel/SatelliteManifestPanel.tsx]: Aha ok yep I went the second route with this, changing it in the hook.

File: src/hooks/useUserPermissions.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
+import { useQuery, UseQueryResult } from 'react-query';
+import Cookies from 'js-cookie';
+import { getConfig, authenticateUser } from '../utilities/platformServices';
+
+interface UserPermissions {
```
[the remark this answers is not in the document]
Comment (joeldmyers) [reply src/hooks/useUserPermissions.ts]: ok changed the hook name, file name, interface name, etc. for this.

File: src/components/SatelliteManifestPanel/SatelliteManifestPanel.tsx (hunk 1 above)
Comment (joeldmyers) [reply src/components/SatelliteManifestPanel/SatelliteManifestPanel.tsx]: yeah so it's just a habit from trying to write pure functions where possible. It's not actually a pure function in this case bc it still relies on components outside the function's scope. So I went ahead and changed it so the user object isn't passed in as a param.

File: src/components/SatelliteManifestPanel/SatelliteManifestPanel.tsx (hunk 2 above)
Comment (joeldmyers) [reply src/components/SatelliteManifestPanel/SatelliteManifestPanel.tsx]: see above; changed so it's not included.

File: src/components/SatelliteManifestPanel/SatelliteManifestPanel.tsx (hunk 3 above)
Comment (joeldmyers) [reply src/components/SatelliteManifestPanel/SatelliteManifestPanel.tsx]: OK I have it called now directly. Also I renamed it to `getManifestPanelColumns`. Not sure if that's the best name; was having trouble thinking of a better one. LMK if you have any thoughts.

Comment (zanewoodfin) [reply src/components/SatelliteManifestPanel/SatelliteManifestPanel.tsx]: I was thinking something like getColumnInfo or getColumnSpecifications, just to convey that it returns information about columns, but not actually the data that will be in the columns.  Also could you move the function down to where the other functions are, below the useState calls and regular variable declarations?

File: src/components/ManifestDetailSidePanel/__tests__/ManifestDetailSidePanel.test.tsx [hunk 7]
```diff
@@ -65,7 +81,40 @@ describe('Manifest Detail Side Panel', () => {
...
-    const { container } = render(<ManifestDetailSidePanel {...props} />);
+    const { container } = render(
+      <QueryClientProvider client={queryClient}>
+        <ManifestDetailSidePanel {...props} />
+      </QueryClientProvider>
+    );
+    expect(container).toMatchSnapshot();
+  });
+
+  it("renders successfully with data, but showing 'administratively disabled' for SCA status when user is not SCA capable", () => {
```
Comment (zanewoodfin) [inline src/components/ManifestDetailSidePanel/__tests__/ManifestDetailSidePanel.test.tsx:92]: I'd omit the first clause here about rendering successfully with data to shorten this up.  The SCA status is what we care about here.

**Review by zanewoodfin — CHANGES_REQUESTED** [review]

**Review by zanewoodfin — APPROVED** [review]
